### ISSUE TITLE
Using the googlemaps api key?

### ISSUE THREAD

**Issue by jlemonz** [nuxt/scripts#313]
### 📚 What are you trying to do?

Hi All,

I am really stuck with this. So i use the demo code 

    <ScriptGoogleMaps
        ref="googleMapsRef"
        :center="query"
        :markers="markers"
        :api-key="googleMapsApiKey"
        class="group w-full h-64 rounded-lg"
        above-the-fold
        @ready="handleReady"
      />

  const googleMapsApiKey = process.env.NUXT_GOOGLE_MAPS_API_KEY;

but this doesnt work? When i use my api key directly in the file it works. But putting it in the .env i wont load the map? I thought placing your api key is not secure? Already asked chatgpt but no solution. 

### 🔍 What have you tried?

I looked at the docs. Nothing about it. I tried chatgpt and different solutions nothing works.

**Comment by manniL** [comment]
You should use [runtimeConfig](https://nuxt.com/docs/api/composables/use-runtime-config)

**Comment by harlan-zw** [comment]
The setup instructions need to be improved for this page. Using `.env` is supported but you need to follow the format of `NUXT_PUBLIC_SCRIPTS_GOOGLE_MAPS_API_KEY`

```ts
// nuxt.config.ts
export default defineNuxtConfig({
  // you need to provide a runtime config to access the environment variables
  runtimeConfig: {
    public: {
      scripts: {
        googleMaps: {
          apiKey: '', // NUXT_PUBLIC_SCRIPTS_GOOGLE_MAPS_API_KEY
        },
      },
    },
  },
})
```

```bash
# .env
NUXT_PUBLIC_SCRIPTS_GOOGLE_MAPS_API_KEY=<your-token>
```

I'll leave open to track doc update

**Comment by jlemonz** [comment]
Thank you i got it working. Is using public not a security issue? I read that I can enable on google cloud what site can use the api key. Or is the key hidden in the script?

**Comment by manniL** [comment]
> Thank you i got it working. Is using public not a security issue? I read that I can enable on google cloud what site can use the api key. Or is the key hidden in the script?

Google Maps needs a public API key. For other APIs you definitely don't want it public (e.g. the stripe **private** key). Talked a bit about that [in this video](https://www.youtube.com/watch?v=2tKOZc3Z1dk)

**Comment by harlan-zw** [comment]
Yes just to add on to this, make sure you lock down your google maps API key to specific hosts!

Very important step https://developers.google.com/maps/api-security-best-practices#restricting-api-keys
